ios/Classes/AdmobBanner.swift

```diff
@@ -1,8 +1,134 @@
+        adView = nil
+        channel.setMethodCallHandler(nil)
+    }
+    
+    fileprivate func getBannerAdView() -> GADBannerView? {
+        if adView == nil {
+            adView = GADBannerView()
+            adView!.rootViewController = UIApplication.shared.keyWindow?.rootViewController
+            adView!.frame = self.frame.width == 0 ? CGRect(x: 0, y: 0, width: 1, height: 1) : self.frame
+            adView!.adUnitID = self.args["adUnitId"] as? String ?? "ca-app-pub-3940256099942544/2934735716"
+            channel.setMethodCallHandler { [weak self] (flutterMethodCall: FlutterMethodCall, flutterResult: FlutterResult) in
+                switch flutterMethodCall.method {
+                case "setListener":
+                    self?.adView?.delegate = self
+                    break
+                case "dispose":
+                    self?.dispose()
+                    break
+                default:
+                    flutterResult(FlutterMethodNotImplemented)
+                }
+            }
+            requestAd()
+        }
+        
+        return adView
+    }
+    
+    fileprivate func requestAd() {
+        if let ad = getBannerAdView() {
+            let request = GADRequest()
+            request.testDevices = [kGADSimulatorID]
+            ad.load(request)
+        }
+    }
+    
+    fileprivate func getSize() -> GADAdSize {
+        let size = args["adSize"] as? [String: Any]
+        let width = size!["width"] as? Int ?? 0
+        let height = size!["height"] as? Int ?? 0
+        let name = size!["name"] as! String
+        
+        switch name {
+        case "BANNER":
+            return kGADAdSizeBanner
+        case "LARGE_BANNER":
+            return kGADAdSizeLargeBanner
+        case "MEDIUM_RECTANGLE":
+            return kGADAdSizeMediumRectangle
+        case "FULL_BANNER":
+            return kGADAdSizeFullBanner
+        case "LEADERBOARD":
+            return kGADAdSizeLeaderboard
+        case "SMART_BANNER":
+            // TODO: Do we need Landscape too?
+            return kGADAdSizeSmartBannerPortrait
+        default:
+            return GADAdSize.init(size: CGSize(width: width, height: height), flags: 0)
+        }
+    }
+
+}
+
+extension AdmobBanner : GADBannerViewDelegate {
+    func adViewDidReceiveAd(_ bannerView: GADBannerView) {
+        channel.invokeMethod("loaded", arguments: nil)
+    }
+    
+    func adView(_ bannerView: GADBannerView, didFailToReceiveAdWithError error: GADRequestError) {
+         channel.invokeMethod("failedToLoad", arguments: error)
+    }
+    
+    /// Tells the delegate that a full screen view will be presented in response to the user clicking on
+    /// an ad. The delegate may want to pause animations and time sensitive interactions.
+    func adViewWillPresentScreen(_ bannerView: GADBannerView) {
+        channel.invokeMethod("clicked", arguments: nil)
+        channel.invokeMethod("opened", arguments: nil)
+    }
```

Comment:
This is one example of iOS not matching up completely with Android. Combining these callbacks to Flutter made sense to me. At least all callbacks will be fired on the Flutter side. @YoussefKababe do you see any problem with this?

Reply:
@kmcgill88 This looks good to me, can't see a problem with it